libr/core/cbin.c

```diff
@@ -1020,6 +1020,38 @@ static int bin_pdb(RCore *core, int mode) {
 	return r_core_pdb_info (core, core->bin->file, baddr, mode);
 }
 
+static int srclineCmp(const void *a, const void *b) {
+	const char *fa = a;
+	const char *fb = b;
+	if (fa && fb) {
+		return !strcmp (fa, fb);
+	}
+	return !strcmp (fa, fb);
+}
+
+static int bin_source(RCore *r, int mode) {
+	RList *final_list = r_list_new ();
+	RBinFile * binfile = r->bin->cur;
+	SdbListIter *iter;
+	RListIter *iter2;
+	char* srcline;
+	SdbKv *kv;
+	SdbList *ls = sdb_foreach_list (binfile->sdb_addrinfo, false);
+	ls_foreach (ls, iter, kv) {
+		char *v = kv->value;
+		RList *list = r_str_split_list (v, "|");
+		if (!strstr (r_list_get_bottom (list), "0x")){
```

Comment:
If for some reason the list has no head, r_list_get_bottom might return NULL. Better use it appart and check its value.